src/server/storage.zig

```diff
@@ -296,71 +302,77 @@ pub const ChunkCompression = struct { // MARK: ChunkCompression
 			const compressedData = main.utils.Compression.deflate(main.stackAllocator, &uncompressedData, .default);
 			defer main.stackAllocator.free(compressedData);
 
-			const data = allocator.alloc(u8, 4 + 1 + 4*ch.data.paletteLength + compressedData.len);
-			std.mem.writeInt(i32, data[0..4], @intFromEnum(CompressionAlgo.deflate_with_8bit_palette), .big);
-			data[4] = @intCast(ch.data.paletteLength);
+			var writer = BinaryWriter.initCapacity(allocator, .big, @sizeOf(CompressionAlgo) + @sizeOf(u8) + @sizeOf(u32)*ch.data.paletteLength + compressedData.len);
+
+			writer.writeEnum(CompressionAlgo, .deflate_with_8bit_palette);
+			writer.writeInt(u8, @intCast(ch.data.paletteLength));
+
 			for(0..ch.data.paletteLength) |i| {
-				std.mem.writeInt(u32, data[5 + 4*i ..][0..4], ch.data.palette[i].toInt(), .big);
+				writer.writeInt(u32, ch.data.palette[i].toInt());
 			}
-			@memcpy(data[5 + 4*ch.data.paletteLength ..], compressedData);
-			return data;
+			writer.writeSlice(compressedData);
+			return writer.data.toOwnedSlice();
 		}
-		var uncompressedData: [chunk.chunkVolume*@sizeOf(u32)]u8 = undefined;
+		var uncompressedWriter = BinaryWriter.initCapacity(main.stackAllocator, .big, chunk.chunkVolume*@sizeOf(u32));
+		defer uncompressedWriter.deinit();
+
 		for(0..chunk.chunkVolume) |i| {
-			std.mem.writeInt(u32, uncompressedData[4*i ..][0..4], ch.data.getValue(i).toInt(), .big);
+			uncompressedWriter.writeInt(u32, ch.data.getValue(i).toInt());
 		}
-		const compressedData = main.utils.Compression.deflate(main.stackAllocator, &uncompressedData, .default);
+		const compressedData = main.utils.Compression.deflate(main.stackAllocator, uncompressedWriter.data.items, .default);
 		defer main.stackAllocator.free(compressedData);
-		const data = allocator.alloc(u8, 4 + compressedData.len);
 
-		@memcpy(data[4..], compressedData);
-		std.mem.writeInt(i32, data[0..4], @intFromEnum(CompressionAlgo.deflate), .big);
-		return data;
+		var compressedWriter = BinaryWriter.initCapacity(allocator, .big, @sizeOf(CompressionAlgo) + compressedData.len);
+
+		compressedWriter.writeEnum(CompressionAlgo,  .deflate);
```

Comment:
```suggestion
		compressedWriter.writeEnum(CompressionAlgo, .deflate);
```

Reply:
What's with all the "suggested change" comments?